elytron/src/main/java/org/wildfly/extension/elytron/FileSystemRealmDefinition.java

```diff
@@ -192,12 +314,19 @@ public void dispose() {
 
             ServiceBuilder<SecurityRealm> serviceBuilder = serviceTarget.addService(mainServiceName, fileSystemRealmService)
                     .addAliases(aliasServiceName);
-
+            if (credentialStore != null) {
+                serviceBuilder.requires(context.getCapabilityServiceName(buildDynamicCapabilityName(CREDENTIAL_STORE_CAPABILITY, credentialStore), CredentialStore.class));
+            }
+            if (keyStoreName != null) {
+                serviceBuilder.addDependency(context.getCapabilityServiceName(
+                        buildDynamicCapabilityName(KEY_STORE_CAPABILITY, keyStoreName), KeyStore.class),
+                        KeyStore.class, keyStoreInjector);
+            }
             if (relativeTo != null) {
                 serviceBuilder.addDependency(PathManagerService.SERVICE_NAME, PathManager.class, pathManagerInjector);
                 serviceBuilder.requires(pathName(relativeTo));
             }
-            serviceBuilder.install();
+            serviceBuilder.addAliases(aliasServiceName).install();
```

Comment:
This should be removed. `addAliases` is called above on line 316.

Reply:
Removed